pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -77,45 +77,85 @@ func New(sess *session.Session, cfg *aws.Config, referentAuth bool) (*SecretsMan
 	}, nil
 }
 
-func (sm *SecretsManager) fetch(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (*awssm.GetSecretValueOutput, error) {
+func (sm *SecretsManager) fetch(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (*awssm.GetSecretValueOutput, error) {
 	ver := "AWSCURRENT"
+	valueFrom := "SECRET"
 	if ref.Version != "" {
 		ver = ref.Version
 	}
-	log.Info("fetching secret value", "key", ref.Key, "version", ver)
+	if ref.MetadataPolicy == esv1beta1.ExternalSecretMetadataPolicyFetch {
+		valueFrom = "TAG"
+	}
+
+	log.Info("fetching secret value", "key", ref.Key, "version", ver, "value", valueFrom)
 
-	cacheKey := fmt.Sprintf("%s#%s", ref.Key, ver)
+	cacheKey := fmt.Sprintf("%s#%s#%s", ref.Key, ver, valueFrom)
 	if secretOut, found := sm.cache[cacheKey]; found {
 		log.Info("found secret in cache", "key", ref.Key, "version", ver)
 		return secretOut, nil
 	}
 
-	var getSecretValueInput *awssm.GetSecretValueInput
-	if strings.HasPrefix(ver, "uuid/") {
-		versionID := strings.TrimPrefix(ver, "uuid/")
-		getSecretValueInput = &awssm.GetSecretValueInput{
-			SecretId:  &ref.Key,
-			VersionId: &versionID,
+	var secretOut *awssm.GetSecretValueOutput
+	var err error
+
+	if ref.MetadataPolicy == esv1beta1.ExternalSecretMetadataPolicyFetch {
+		describeSecretInput := &awssm.DescribeSecretInput{
+			SecretId: &ref.Key,
+		}
+
+		descOutput, err := sm.client.DescribeSecretWithContext(ctx, describeSecretInput)
+		if err != nil {
+			return nil, err
+		}
+		log.Info("found metadata secret", "key", ref.Key, "output", descOutput)
+
+		secretOut = &awssm.GetSecretValueOutput{
+			ARN:          descOutput.ARN,
+			CreatedDate:  descOutput.CreatedDate,
+			Name:         descOutput.Name,
+			SecretString: TagsToJSONString(descOutput.Tags),
+			VersionId:    &ver,
 		}
 	} else {
-		getSecretValueInput = &awssm.GetSecretValueInput{
-			SecretId:     &ref.Key,
-			VersionStage: &ver,
+		var getSecretValueInput *awssm.GetSecretValueInput
+		if strings.HasPrefix(ver, "uuid/") {
+			versionID := strings.TrimPrefix(ver, "uuid/")
+			getSecretValueInput = &awssm.GetSecretValueInput{
+				SecretId:  &ref.Key,
+				VersionId: &versionID,
+			}
+		} else {
+			getSecretValueInput = &awssm.GetSecretValueInput{
+				SecretId:     &ref.Key,
+				VersionStage: &ver,
+			}
+		}
+		secretOut, err = sm.client.GetSecretValue(getSecretValueInput)
+		var nf *awssm.ResourceNotFoundException
+		if errors.As(err, &nf) {
+			return nil, esv1beta1.NoSecretErr
+		}
+		if err != nil {
+			return nil, err
 		}
-	}
-	secretOut, err := sm.client.GetSecretValue(getSecretValueInput)
-	var nf *awssm.ResourceNotFoundException
-	if errors.As(err, &nf) {
-		return nil, esv1beta1.NoSecretErr
-	}
-	if err != nil {
-		return nil, err
 	}
 	sm.cache[cacheKey] = secretOut
 
 	return secretOut, nil
 }
 
+func TagsToJSONString(tags []*awssm.Tag) *string {
+	jsonString := "{"
+	for _, tag := range tags {
+		jsonString += "\"" + *tag.Key + "\":\"" + *tag.Value + "\","
+	}
```

Comment:
My first impression reading this was that this may be a bit brittle concatenating strings to construct a json string.
[Looking at the docs](https://docs.aws.amazon.com/mediaconnect/latest/ug/tagging-restrictions.html) it seems that only alphanumeric characters plus ` . : + = @ _ / - ` are accepted values for both keys and values.
If AWS decides to support other characters (e.g. `\` or `"`) then this will likely break.
This isn't a issue now, but it could become one in the future.

Maybe it would be better to populate a `map[string]string` and marshal it?

Reply:
It was actually a red flag to me and I was hoping it would be raised here... I ment to mention it but I forgot. I knew there should be a better way, but forgot all about marchalling a dictionary. Thanks for the suggestion, will change it